packages/ShowMore/src/ShowMore.js

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import ReactDOMServer from "react-dom/server";
+import PropTypes from "prop-types";
+import truncate from "lodash/truncate";
+import uuid from "uuid/v4";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import Button from "@paprika/button";
+import { toggleStyles } from "./ShowMore.styles";
+
+const propTypes = {
+  /** Additional description for "Show more" link. Should be a "topic" that will be appended to "Show more about [topic]". */
+  a11yText: PropTypes.string,
+
+  /* Full content to be revealed. */
+  children: PropTypes.node.isRequired,
+
+  /* Length, in characters, of truncated preview content. */
+  collapsedLength: PropTypes.number,
+};
+
+const defaultProps = {
+  a11yText: null,
+  collapsedLength: 255,
+};
+
+function getTruncatedVersion(content, length) {
+  return truncate(content, {
+    length,
+    separator: " ",
+  });
+}
+
+const ShowMore = props => {
+  const { a11yText, children, collapsedLength, ...moreProps } = props;
+
+  const I18n = useI18n();
+  const [isCollapsed, setIsCollapsed] = React.useState(true);
+  const [collapsedContent, setCollapsedContent] = React.useState();
+  const contentId = React.useRef(uuid()).current;
+
+  React.useEffect(() => {
+    setCollapsedContent(`${getTruncatedVersion(ReactDOMServer.renderToStaticMarkup(children), collapsedLength)} `);
+  }, [children, collapsedLength]);
```

Comment:
move `getTruncatedVersion` inside of the use effect because is a dependency, you can't add it because will make your component rerender all the time. 

you eslint should be warning you about this.

Reply:
I think in this case it's fine because `getTruncatedVersion` doesn't depend on any props or state and can safely be created outside of the component altogether.
(As an absurd example, the same can be said about `ReactDOMServer.renderToStaticMarkup`—you're not specifying it as a dependency, right?)

Reply:
(find "You can try moving that function outside of your component" in this section: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies)

Reply:
But ACKCHUALLY I have a couple issues with this effect as is:
- I don't think this needs to be an effect at all? It's not a side effect and it doesn't depend on the DOM being updated. So instead of having collapsed content as state, it can just be inlined into `dangerouslySetInnerHTML` (`__html: ReactDOM.renderToStaticMarkup(...)`)
- What's the current use case for this? `renderToStaticMarkup` assumes server-rendering compatibility of what is passed into it. Meaning you generally can't pass arbitrary component tree and expect it to work. (notably, if a component needs its `componentDidMount` invoked to be able to work properly, that will not fly) So this component will be restricted to simpler use cases (e.g. paragraphs of static content), which I think should be fine?